01_fibonacci/fib.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fibonacci(10)
+}
+
+func fibonacci(n int) {
+
+	var sum int
+	var prev, cur int = 0, 1
+	var fibSeries string
+
+	for i := 1; i <= n; i++ {
+		// fmt.Println(prev)
+		fibSeries += strconv.Itoa(prev) + " "
+		sum = prev + cur
+		prev = cur
+		cur = sum
```

Comment:
You can also do a one-line here: `prev, cur = cur, prev + cur`.

Reply:
Yes, correct, but somehow I get weird output if I put one-liner

Reply:
This is unnecessarily complicated:
```go
	a, b := 1, 1
	for i := 1; i <= n; i++ {
		fmt.Fprintln(a)
		a, b = b, a+b
	}
```
should do the trick.
If you want to collect values, collect them in a slice and return it. Let main handle the printing....

Reply:
> if you want to collect values, collect them in a slice and return it. Let main handle the printing

i agree! https://github.com/anz-bank/go-training/issues/37

Reply:
Corrected with right approach.